### ISSUE TITLE
[Bug]: (master) Login flow currently broken in old-ish Android versions

### ISSUE THREAD

**Issue by AlvaroBrey** [nextcloud/server#36934]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When trying to log in on mobile with current server master, the login "Connect to your account" screen is unresponsive: the "Log in" button remains colorless and does nothing when tapped. 

HTML inspection reveals the `input` is `disabled`. Screenshot:

<details>

![Screenshot_1677666576](https://user-images.githubusercontent.com/8677743/222116811-010a7a4c-cdf0-4dda-999d-225b4725b83c.png)

</details>

### Steps to reproduce

1. Run an instance of master server (I tested with https://github.com/szaimen/nextcloud-easy-test, and with continuous-integration/shallow-server)
2. Try to log in with the Android app
3. See unresponsive "Connect to your account" screen

### Expected behavior

I expected the Log in button to do something :(

### Installation method

Other Community project

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

SQlite

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "toaster.lan"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "sqlite3",
        "version": "26.0.0.7",
        "overwrite.cli.url": "http:\/\/localhost",
        "updater.release.channel": "git",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "log_type": "errorlog",
        "maintenance": false
    }
}
```

### List of activated Apps

```shell
Enabled:
  - cloud_federation_api: 1.9.0
  - comments: 1.16.0
  - contactsinteraction: 1.7.0
  - dashboard: 7.6.0
  - dav: 1.25.0
  - federatedfilesharing: 1.16.0
  - federation: 1.16.0
  - files: 1.21.1
  - files_sharing: 1.18.0
  - files_trashbin: 1.16.0
  - files_versions: 1.19.1
  - lookup_server_connector: 1.14.0
  - oauth2: 1.14.0
  - photos: 2.2.0
  - provisioning_api: 1.16.0
  - settings: 1.8.0
  - sharebymail: 1.16.0
  - systemtags: 1.16.0
  - text: 3.7.2
  - theming: 2.1.1
  - twofactor_backupcodes: 1.15.0
  - updatenotification: 1.16.0
  - user_status: 1.6.0
  - viewer: true
  - weather_status: 1.6.0
  - workflowengine: 2.8.0
Disabled:
  - admin_audit: 1.16.0
  - encryption: 2.14.0
  - files_external: 1.18.0
  - testing: 1.16.0
  - user_ldap: 1.16.0
```

### Nextcloud Signing status

```shell
Integrity checker has been disabled. Integrity cannot be verified.
```

### Nextcloud Logs

```shell
Log is empty
```

### Additional info

[console-1677666558746.log](https://github.com/nextcloud/server/files/10860005/console-1677666558746.log)

**Comment by szaimen** [comment]
cc @SystemKeeper @nickvergessen

**Comment by SystemKeeper** [comment]
Hm, why is jQuery not defined? 

```
authpicker.js?v=9169f930-0:1 Uncaught ReferenceError: jQuery is not defined
    at authpicker.js?v=9169f930-0:1
```

**Comment by szaimen** [comment]
Not sure however I think this is likely cause by https://github.com/nextcloud/server/pull/36286

**Comment by SystemKeeper** [comment]
The result yes, because the input is disabled. But jQuery was there before 🤔

**Comment by nickvergessen** [comment]
loading order?
Then again
```
Query(document).ready(function() {
```

should do it (and its there already)

**Comment by SystemKeeper** [comment]
I’ll try to set up an android device later and take a look.
I’m not seeing a reason it fails only on android

**Comment by AlvaroBrey** [comment]
Here's the console log for the same android device in 25: 
[toaster.lan-1677670100465.log](https://github.com/nextcloud/server/files/10860425/toaster.lan-1677670100465.log)

The JS load errors are not present

**Comment by AlvaroBrey** [comment]
> I’ll try to set up an android device later and take a look. I’m not seeing a reason it fails only on android

For reference we're using an Android Webview (not a regular browser) to load the login flow. If you need any help debugging on the Android side contact me on Talk

**Comment by nickvergessen** [comment]
Is it a persistent issue? Can you reproduce it sometimes or always?

**Comment by SystemKeeper** [comment]
Would also make sense to test Beta3 since it’s the first beta to include the changes. To rule out a regression introduced later

**Comment by AlvaroBrey** [comment]
> Is it a persistent issue? Can you reproduce it sometimes or always?

Always

**Comment by SystemKeeper** [comment]
Okay, debugged this one and confirmed with Alvaro. It happens on older versions of the web view (so older android version) which do not support null coalescing operators (??) in javascript. This is only supported on web view versions >= 80.

The PR https://github.com/nextcloud/server/pull/36813 introduced that.

f32804e97defd7f231377117fcbe758ab5cf0239:
![Bildschirm­foto 2023-03-01 um 14 05 18](https://user-images.githubusercontent.com/1580193/222147569-ae656c21-0df3-478e-8ac5-b11901be0709.png)

12571079fc90d00dcc74b3a0f953b853a854b969:
![image](https://user-images.githubusercontent.com/1580193/222147817-095f0c83-d72c-4a56-9ddf-e7de9af9efe7.png)

@Pytal Any idea about this?

**Comment by Pytal** [comment]
Which version of Android is used @SystemKeeper?

We target Android 4.4.3-4.4.4 WebView in our browserslist https://browserslist.dev/?q=PjAuMjUlLCBub3QgaWUgMTEsIG5vdCBvcF9taW5pIGFsbCwgbm90IGRlYWQsIEZpcmVmb3ggRVNS which is based on Chrome 33 https://developer.chrome.com/docs/multidevice/webview/#what-version-of-chrome-is-it-based-on

**Comment by nickvergessen** [comment]
Android 6 is using Chrome 80 as far as I understood 🥲🙈

**Comment by Pytal** [comment]
Also, it does not seem to be the `??` nullish coalescing operator as they are present on `stable24` as well https://github.com/nextcloud/server/blob/stable24/dist/core-common.js

**Comment by SystemKeeper** [comment]
> Also, it does not seem to be the `??` nullish coalescing operator as they are present on `stable24` as well https://github.com/nextcloud/server/blob/stable24/dist/core-common.js

Maybe it wasn't a problem back there, because we didn't need any jQuery support here. It was a normal link in NC 24:
https://github.com/nextcloud/server/blob/stable24/core/templates/loginflow/authpicker.php#L48-L52

It was changed here: https://github.com/nextcloud/server/commit/3fa6ee3c605e097c8a026a173c39679cf7681045

But I don't see the use of the `??` operator in this file?!
![image](https://user-images.githubusercontent.com/1580193/222250439-d9a6d779-7c5a-4b6b-8dea-6d5cb9f12073.png)

**Comment by SystemKeeper** [comment]
![image](https://user-images.githubusercontent.com/1580193/222254926-90b424d5-f0a7-41f8-90dc-ad85b6c715d5.png)

What I don't understand, if we include 4.4.3-4.4.4, then there should be no null-coalescing operator according to this:
https://caniuse.com/mdn-javascript_operators_nullish_coalescing
![image](https://user-images.githubusercontent.com/1580193/222255623-78f5bd4a-8c0f-4d5b-a5ff-8e032e650a6f.png)

Android webview earliest version 80 supports this, if 4.4.3 is based on Chrome 33, this shouldn't be there?!

**Comment by AlvaroBrey** [comment]
Chrome version is unrelated to android webview version AFAIK.  Either way the device I'm testing with is android SDK 28 (Android 9) and it comes with Webview 66.

**Comment by Pytal** [comment]
> But I don't see the use of the `??` operator in this file (The NC24 one you linked)?! ![image](https://user-images.githubusercontent.com/1580193/222250439-d9a6d779-7c5a-4b6b-8dea-6d5cb9f12073.png)

😅 right, did not realize that it's actually a regex

> Android webview earliest version 80 supports this, if 4.4.3 is based on Chrome 33, this shouldn't be there?!

Yeah, very strange 🤔

So does the commit right before bfad19c09d6a50ec6156eb344dddf38332f62875 work? While 12571079fc90d00dcc74b3a0f953b853a854b969 doesn't @SystemKeeper @AlvaroBrey?

**Comment by SystemKeeper** [comment]
```
www-data@76e1396ef5da:~/nextcloud$ git status
HEAD detached at bfad19c09d
```

-> No syntax error (the styling is a bit off, but I think that's unrelated here, at least I can click on "Log In")


```
www-data@76e1396ef5da:~/nextcloud$ git status
HEAD detached at 12571079fc
```

![image](https://user-images.githubusercontent.com/1580193/222287092-29ac5a9f-4013-401d-a34a-22a5fd89c2a3.png)

**Comment by SystemKeeper** [comment]
When I manually install 7.6.0 of the vue lib (`npm install @nextcloud/vue@7.6.0`) and then run `make build-js`, I don't see the `??` operator anymore. So something between 7.6.0 and 7.7.0 changed.

**Comment by SystemKeeper** [comment]
Looks like this comes from @nextcloud/axios through @nextcloud/vue-richtext.
![image](https://user-images.githubusercontent.com/1580193/222290391-0bff2548-ee43-4969-8036-b16af0dfc13c.png)

**Comment by blizzz** [comment]
fixed by #37005
